### PR TITLE
Issue #8523: Records support check update for EmptyLineSeparatorCheck

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #8523: Records support check update for EmptyLineSeparatorCheck

Diff Regression projects: https://gist.githubusercontent.com/nmancus1/e3988f8092d919b5cbe70f5359c4d9e8/raw/a0f6f5860f8025c19868061dee6e0e40960b992f/projects-to-test-on.properties

Diff Regression config: https://gist.githubusercontent.com/nmancus1/bdaeb2e6be6d30243edaeb6f04e44a87/raw/bc4eff061891f13d2ec1cf2beeab87fd96e4cc93/EmptyLineSeparatorCheck.xml

### CODE REVIEW

**Comment by nmancus1** [comment]
Github, generate report

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorRecordsAndCompactCtors.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.whitespace.emptylineseparator; // violation
```
Comment (pbludov) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorRecordsAndCompactCtors.java:2]: This is not related to the issue. Will this violation gone if you add an empty line between the package declaration and the header?

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorRecordsAndCompactCtorsNoEmptyLines.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.whitespace.emptylineseparator; // violation
```
Comment (pbludov) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorRecordsAndCompactCtorsNoEmptyLines.java:2]: Same here

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorRecordsAndCompactCtors.java (hunk 1 above)
Comment (nmancus1) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorRecordsAndCompactCtors.java:2]: Yes, the violation will be gone if these lines are separated? Should they be? This is the format I've followed with all other inputs, so I left it even though it was unrelated, for consistency.

Comment (pbludov) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorRecordsAndCompactCtors.java:2]: I'm OK to leave it as is.
